django/db/migrations/operations/models.py

```diff
@@ -802,6 +805,7 @@ def state_forwards(self, app_label, state):
         model_state = state.models[app_label, self.model_name_lower]
         indexes = model_state.options[self.option_name]
         model_state.options[self.option_name] = [idx for idx in indexes if idx.name != self.name]
+        state.reload_model(app_label, self.model_name_lower, delay=True)
```

Comment:
I tried adding these lines to the existing test but they're passing:
```python
operation = migrations.RemoveIndex('Pony', index.name)
end_state = new_state.clone()
operation.state_forwards('test_adin', end_state)
before_indexes = new_state.models['test_adin', 'pony'].options['indexes']
self.assertEqual(before_indexes, [index])
after_indexes = end_state.models['test_adin', 'pony'].options['indexes']
self.assertEqual(after_indexes, [])
```
so I'm not sure what issue this line is solving. Also, seems like the test should go in `test_state.py`?